### DISCUSSION TITLE
Using multiple lua versions in the same project

### DISCUSSION THREAD

**Discussion by toxamin** [mlua-rs/mlua#321] [Q&A] [answered]
Is it possible to use multiple versions of lua in the same project using mlua?

**Comment by alerque** [comment]
Not easily due to Cargo's feature unification (which they consider a feature not a bug), but probably if you wanted to setup workspaces with separate crates for each variant, then import then into a project with various names...

The question is *why*?

**Reply by toxamin** [reply]
I have a need to process files with different lua versions in the same project, and must interact with the api while doing so.

**Comment by khvzak** [comment, marked as answer]
You can do this, probably, by creating a few dynamic libraries, one for each lua version.
Then load required libraries (all or some) using [libloading](http://crates.io/crates/libloading).
It's not trivial unfortunately (you basically need to write plugins for your app). Take a look to [abi_stable](https://crates.io/crates/abi_stable) as well.
